src/pudl/constants.py

```diff
@@ -2771,3 +2786,23 @@
         'zip_code': pd.StringDtype(),
     },
 }
+
+ENTITY_TYPE_DICT = {
+    'M': 'Municipal',
+    'C': 'Cooperative',
+    'R': 'Retail Power Marketer',
+    'I': 'Investor Owned',
+    'P': 'Political Subdivision',
+    'T': 'Transmission',
+    'S': 'State',
+    'W': 'Wholesale Power Marketer',
+    'F': 'Federal',
+    'A': 'Municipal Mktg Authority',
+    'G': 'Community Choice Aggregator',
+    'D': 'Nonutility DSM Administrator',
+    'B': 'Behind the Meter',
+    'Q': 'Independent Power Producer',
+    'IND': 'Industrial',
+    'COM': 'Commercial',
+    float('nan'): 'Unregulated',
```

Comment:
Is there some way that the values here can be used directly in the definition of the `CategoricalDtype()` for `entity_type` -- otherwise we're storing this list of valid values in two places, and it will be easy to forget to update both of them if they change.

Reply:
Good point. I'll check on that. The nan value comes from the `Frame` table. This table has both the "code" and the "full" ownership columns. All of the nan values in the "code" column showed up as "unregulated" in the "full" ownership column. This wasn't, however, specified in the instructions, so I can easily take it out.

Reply:
I'm not sure it should be taken out, it just seemed like an odd case, and I wonder if it really applies across the board. Like IPPs are typically unregulated too, but they get their own code. And do the NaN values that we might encounter in other tables also necessarily correspond to unregulated?

Reply:
right...not sure what is best in this case.